src/lib/y2network/interface_config_builder.rb

```diff
@@ -343,13 +343,16 @@ def hostname=(value)
       @hostname = value
     end
 
+    # Saves the hostname
+    #
+    # It needs to take into account whether the old configuration and the boot protocol.
     def save_hostname
-      # avoid unnecessary modification
-      return if @original_ip_config == @connection_config.ip && @original_hostname == hostname
-
-      # remove old ip
-      Yast::Host.remove_ip(@original_ip_config.ip.address.to_s) if @original_ip_config != @connection_config.ip
+      if !required_ip_config? || hostname.empty?
```

Comment:
are you sure that it covers case when ip is different and hostname is different? I worry this handle only case when static to dynamic or hostname delete. And in fact Update handles also empty hostname and properly remove it. I think we definitivelly need to document all possible cases.

Reply:
I've added some test cases and fixed a few issues because even the comparisons where not working at all.